src/components/CommentCard.jsx

```diff
@@ -1,10 +1,32 @@
+import { deleteComment } from "./utils/api"
+const CommentCard = ({ comment, setCommentsOnArticle }) => {
+    const username = "tickle122"
 
-const CommentCard = ({comment}) => {
+    const handleClick = (comment_id, comment_author) => {
+        if (comment_author !== username) {
+            alert('You can only delete your own comments!')
+            return
+        }
+        setCommentsOnArticle((currComments) => {
+            const ammendedComments = [...currComments]
+            for (const comment of ammendedComments) {
+                if (comment.comment_id === comment_id) {
+                    const index = ammendedComments.indexOf(comment)
+                    ammendedComments.splice(index, 1)
+                    return ammendedComments
+                }
+            }
+        })
+        return deleteComment(comment_id).then(() => {
+            alert('Comment deleted...')
```

Comment:
You will want to set the state in the then block. Otherwise you'll have yo un-do the state setting in a catch block.